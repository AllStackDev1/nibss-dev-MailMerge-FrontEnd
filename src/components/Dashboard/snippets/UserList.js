import React from 'react'
import EmptyUser from '../empty-states/EmptyUser'
import Pagination from './Pagination'
import User from './User'

const UserList = ({ users, search, filter, setModal, setUser, initiateDeleteUser, updateRole, viewPage }) => {
    const renderUsers = () => {
        if (users?.platformUsers === undefined || (users.searching)) {
            return <EmptyUser />;
        }

        const toLoop = (search.search !== "" || filter !== false) && users.searchResults ?
            users.searchResults :
            users.platformUsers;

        return <>
            {(toLoop).data.map((u, index) =>
                <User
                    key={index}
                    setModal={setModal}
                    setUser={setUser}
                    deleteUser={initiateDeleteUser}
                    updateRole={updateRole}
                    userBeingUpdated={users.deleting || users.updatingRole}
                    user={u} />
            )}
            <Pagination
                data={toLoop}
                viewPage={viewPage}
            />
        </>
    }

    return renderUsers()
}

export default UserList;
