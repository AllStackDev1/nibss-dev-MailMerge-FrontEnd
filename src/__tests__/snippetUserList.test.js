import React from 'react';
import { mount } from 'enzyme';
import UserList from "../components/Dashboard/snippets/UserList"


const shallowSetup = (props = {}) => {

    return mount(
        <UserList {...props} />
    )
}

it("should render properly", () => {

    const wrapper = shallowSetup({ users: { platformUsers: "123", searching: true, searchResults: "result", platformUsers: "users" } });
    expect(wrapper).toMatchSnapshot();

})


