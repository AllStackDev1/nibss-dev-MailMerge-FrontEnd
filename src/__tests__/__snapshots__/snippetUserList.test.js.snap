// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly with searching false 1`] = `ReactWrapper {}`;

exports[`should render properly with searching true 1`] = `ReactWrapper {}`;
